worker/src/RTC/SCTP/chunkParameters/CookiePreservativeChunkParameter.cpp

```diff
@@ -0,0 +1,148 @@
+#define MS_CLASS "RTC::SCTP::CookiePreservativeChunkParameter"
+// #define MS_LOG_DEV_LEVEL 3
+
+#include "RTC/SCTP/chunkParameters/CookiePreservativeChunkParameter.hpp"
+#include "Logger.hpp"
+#include "MediaSoupErrors.hpp"
+
+namespace RTC
+{
+	namespace SCTP
+	{
+		/* Class methods. */
+
+		CookiePreservativeChunkParameter* CookiePreservativeChunkParameter::Parse(
+		  const uint8_t* buffer, size_t bufferLength)
+		{
+			MS_TRACE();
+
+			ChunkParameter::ChunkParameterType parameterType;
+			uint16_t parameterLength;
+			uint8_t padding;
+
+			if (!ChunkParameter::IsChunkParameter(
+			      buffer, bufferLength, parameterType, parameterLength, padding))
+			{
+				return nullptr;
+			}
+
+			if (parameterType != ChunkParameter::ChunkParameterType::COOKIE_PRESERVATIVE)
+			{
+				MS_WARN_DEV("invalid Chunk Parameter type");
+
+				return nullptr;
+			}
+
+			return CookiePreservativeChunkParameter::ParseStrict(
+			  buffer, bufferLength, parameterLength, padding);
+		}
+
+		CookiePreservativeChunkParameter* CookiePreservativeChunkParameter::ParseStrict(
```

Comment:
`I'm` trying to understand the rationale for this name. Is the process always `Parse()` calling `ParseStrict()`?

Reply:
`Xxxxx::Parse()` is intended for directly parsing a buffer containing `Xxxxx` packet/chunk/parameter/error-cause. We don't really need it in production runtime since what we parse is only received full SCTP `Packets`. But anyway those `Xxxxx::Parse()` are useful for testing and maybe needed in the future in case we need to parse full chunks or chunk parameters contained in the body of some SCTP error causes, etc.

`Xxxxx::ParseStrict()` is used within `Packet::Parse()` to parse chunks, chunk parameters and error causes within a SCTP buffer that contains a SCTP packet. It's more efficient the `Xxxxx::Parse()`. In fact, `Xxxxx::Parse()` internally calls `Xxxxx::ParseStrict()`. See any class to understand the difference.

Reply:
IMHO we could name both methods `Parse()`, since they have different arguments we would just use function overloading. I personally don't see a reason behind the name `strict`.

Reply:
After long call, we will keep `ParseStrict()` naming becase it's an "internal" static method and the alternative could be `UnsafeParse()` which sounds terrifying.

Reply:
Note, I've made `ParseStrict()` static methods private because friend class can access them.